### DISCUSSION TITLE
DaCy for Rstudio?

### DISCUSSION THREAD

**Discussion by Klafnorc** [centre-for-humanities-computing/DaCy#289] [Q&A]
Greetings,

Im currently working with spacy in python and spacyr in Rstudio, trying to figure out which is the easies to work with. I cant seem to get Dacy to work in PyCharm or VS and now ive got spacyr to work in Rstudio. So my question is - Do you guys know if theres any way to get DaCy to work with Rstudio?

**Comment by KennethEnevoldsen** [comment]
Hi @Klafnorc! Thanks for your interest in this work.

To clarify a few points: RStudio is a great IDE intended for R, but it does run python script directly. So DaCy works within RStudio, but I assume you refer to R specifically.

DaCy is developed for Python you can also run it within R using packages like reticulate (which spacyr use). However, these are translational software, which are often quite complex or lead to limited usability.

I would generally say that it is well worth your time to spend some time in python to extract the relevant information from the text and export it such that you can work with R for the rest of the workflow. 

If you can describe what you intend to do I don't mind helping you in the right direction.

**Reply by Klafnorc** [reply]
Hi @KennethEnevoldsen, thanks for getting back to me so quickly and for the great explanation.

I am currently using a Python environment in RStudio, mainly because I have encountered some issues configuring PyCharm, but my end goal is to move everything into Python.

What I intend to do is perform sentiment analysis on every netbank message we receive at the bank where I work, and generally work with NLP. Furthermore, I want to support this with some kind of neural network (either CNN or BERT). From what I understand, SpaCy is already built on CNN, correct?

I also see that DaCy is built on the SpaCy-experimental package. Do you know if that's supported in RStudio using reticulate?

All of this is research for my upcoming bachelor's project, just to see what I'm able to do :-)

**Reply by KennethEnevoldsen** [reply]
> What I intend to do is perform sentiment analysis on every netbank message we receive at the bank where I work, and generally work with NLP. Furthermore, I want to support this with some kind of neural network (either CNN or BERT). From what I understand, SpaCy is already built on CNN, correct?

The spacy models can be built on a variety of architectures. The common ones they provide are attention-based CNNs and the ones in Dacy's models are all based on BERT (and there is also a rule-based sentiment model powered by [asent](https://kennethenevoldsen.github.io/asent/introduction.html)), which allows for more interpretability.

> I also see that DaCy is built on the SpaCy-experimental package. Do you know if that's supported in RStudio using reticulate?

I do not, but as far as I understand reticulate just wraps a python session inside of R so you should be good to go.

**Reply by Klafnorc** [reply]
Amazing thank you so much @KennethEnevoldsen. I'll give it a try.

On another note, as a python beginner (ive primarily used Rstudio and R) do you have any suggestions to who to watch or what to read to be proberly set up with python and PyCharm?

**Reply by KennethEnevoldsen** [reply]
I haven't used PyCharm before - have heard good things though - but I personally use VS Code, which my students also use. There is plenty of good videos on youtube for VS code at least so I think that should get you of to a good start.
